### ISSUE TITLE
Chai 4.2.1 incorrectly extends superagent

### ISSUE THREAD

**Issue by pburkindine** [chaijs/chai-http#236]
```
> tslint --project tsconfig.json

node_modules/chai-http/types/index.d.ts:51:19 - error TS2430: Interface 'ChaiHttp.Response' incorrectly extends interface 'import("/Users/pete.burkindine/<...>/node_modules/@types/superagent/index").Response'.
  Types of property 'setEncoding' are incompatible.
    Type '(encoding: string) => void' is not assignable to type '(encoding: string) => this'.
      Type 'void' is not assignable to type 'this'.

51         interface Response extends request.Response {}
                     ~~~~~~~~

node_modules/chai-http/types/index.d.ts:52:19 - error TS2430: Interface 'Agent' incorrectly extends interface 'SuperAgentStatic'.
  Types of property 'del' are incompatible.
    Type '(url: string, callback?: ((err: any, res: Response) => void) | undefined) => Request' is not assignable to type '(url: string, callback?: CallbackHandler | undefined) => SuperAgentRequest'.
      Types of parameters 'callback' and 'callback' are incompatible.
        Types of parameters 'res' and 'res' are incompatible.
          Type 'ChaiHttp.Response' is not assignable to type 'import("/Users/pete.burkindine/<...>/node_modules/@types/superagent/index").Response'.
            Types of property 'setEncoding' are incompatible.
              Type '(encoding: string) => void' is not assignable to type '(encoding: string) => Response'.
                Type 'void' is not assignable to type 'Response'.

52         interface Agent extends request.SuperAgentStatic {
                     ~~~~~


Found 2 errors.
```

**Comment by ahisley** [comment]
I ran into this exact error trace while bumping my chai-http version recently.  While debugging, I noticed that a more recent version of chai-http now ships with type definitions included in the library.  The project where I had these errors was previously set up to use chai-http type definitions from DefinitelyTyped (https://www.npmjs.com/package/@types/chai-http).

For me it was sufficient to remove the (now extraneous) type definitions from DefinitelyTyped.  Once I did that, this error went away.

Might be worth checking if your project is pulling in type definitions from two sources like mine was.

**Comment by RossyFranca** [comment]
I had the same problem, if using typescript and @ types / chai-http just uninstall the chai-http
npm uninstall --save chai-http
and be happy

**Comment by austince** [comment]
Sorry to have not gotten to this until now, but yes, @ahisley is correct - we have added TypeScript definitions in [4.2.0](https://github.com/chaijs/chai-http/releases/tag/4.2.0), though they were not included until [4.2.1](https://github.com/chaijs/chai-http/releases/tag/4.2.1). The types from `@types/chai-http` are outdated and will conflict with the now-included types. We released this as a minor feature as it doesn't break compatibility within `chai-http`, but could have made it more clear that it will conflict with the `@types` package many TypeScript users are probably using. Does removing `@types/chai-http` work for you @pburkindine, or is this potentially something else?
